src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/TestSuiteStepsFlow.tsx

```diff
@@ -0,0 +1,106 @@
+import {useEffect, useMemo, useState} from 'react';
+import ReactFlow, {Controls, Edge, Node} from 'reactflow';
+import 'reactflow/dist/style.css';
+
+import {nanoid} from '@reduxjs/toolkit';
+
+import {LocalStep} from '@src/models/testSuite';
+
+import IntersectionNode from './Nodes/IntersectionNode';
+import StepNode from './Nodes/StepNode';
+import {ReactFlowContainer} from './SettingsTests.styled';
+
+interface ExtendedNode extends Node {
+  group?: number | string;
+}
+
+type TestSuiteStepsFlowProps = {
+  steps: LocalStep[][];
+  setSteps: (steps: LocalStep[][]) => void;
+  showModal: (value: string, group: number | string) => void;
+};
+
+const TestSuiteStepsFlow: React.FC<TestSuiteStepsFlowProps> = props => {
+  const {steps, setSteps, showModal} = props;
+
+  const [nodes, setNodes] = useState<ExtendedNode[]>([]);
+  const [edges, setEdges] = useState<Edge[]>([]);
+
+  const removeNode = (id: string, group: number) => {
+    if (steps[group].length === 1) {
+      setSteps([...steps.slice(0, group), ...steps.slice(group + 1)]);
+    } else {
+      setSteps([...steps.slice(0, group), steps[group].filter(item => item.id !== id), ...steps.slice(group + 1)]);
+    }
```

Comment:
In theory it may have some unexpected bug, when i.e. there is group 1 with single item with ID 1, and it's called with `removeNode(2, 1)`. It would delete the group, despite non-existent ID.

Optionally, to make sure it won't happen:
```suggestion
    const newGroup = steps[group].filter(item => item.id !== id);
    const insertion = newGroup.length === 0 ? [] : [newGroup];
    setSteps([...steps.slice(0, group), ...insertion, ...steps.slice(group)]);
```

But it's optional, as these functions may anyway throw errors (i.e. with not existing group), so not sure if it's worth to fix such single trivial problem, when multiple will stay anyway.

Reply:
This code doesn't work, I will add one more check to prevent removing wrong id